debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -127,6 +127,7 @@ public StreamingResult execute(ChangeEventSourceContext context, SqlServerTaskPa
                 shouldIncreaseFromLsn = offsetContext.getStreamingExecutionState().getShouldIncreaseFromLsn();
             }
             else {
+                tablesSlot = new AtomicReference<SqlServerChangeTable[]>(getCdcTablesToQuery(offsetContext, partition));
```

Comment:
Logic-wise looks right. I'd still like to clarify a few things:
1. Should the atomic reference be initialized here or instead it could be initialized at the beginning of `try` and `set()` here?
2. Do we really need one atomic reference in the streaming execution state and another one here?
3. Why is an atomic reference used here at all? AFAIK, there are no concurrently running threads that would use the same table slot.

Reply:
Given that the `AtomicReference` object itself gets rewritten in line 121 I think we can safely remove it.